polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaInterpreter.scala

```diff
@@ -67,6 +67,9 @@ class ScalaInterpreter(
   ): IO[Stream[IO, Result]] = {
     val originalOut = System.out
 
+    // TODO: this is wrong. It doesn't seem good to keep parsing this from "CellXX" though. Maybe cells also (or only?) need a numeric ID?
```

Comment:
Agreed, I have also been thinking that cells need their own ID space thats separate from position in the notebook and execution order.